cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -206,6 +206,7 @@ void WholeStageResultIterator::setConfToQueryContext(const std::shared_ptr<velox
   std::unordered_map<std::string, std::string> configs = {};
   // Find batch size from Spark confs. If found.
   configs[velox::core::QueryConfig::kPreferredOutputBatchRows] = getConfigValue(kSparkBatchSize, "4096");
+  configs[velox::core::QueryConfig::kMaxOutputBatchRows] = getConfigValue(kSparkBatchSize, "4096");
```

Comment:
It could be considered to use a Spark config if needed in the future.

Reply:
Thanks,  we can fix it when we refactor config module.

Reply:
We definitely need to make it configurable. We have 4 configs now, looks none of them named the config here correctly . Let's change spark.gluten.sql.columnar.shuffleSplitDefaultSize to spark.gluten.sql.columnar.maxBatchSize

  --conf spark.sql.parquet.columnarReaderBatchSize=${batchsize} \
  --conf spark.sql.inMemoryColumnarStorage.batchSize=${batchsize} \
  --conf spark.sql.execution.arrow.maxRecordsPerBatch=${batchsize} \
  --conf spark.gluten.sql.columnar.shuffleSplitDefaultSize=${batchsize} \

Reply:
With above logic, below Spark config is used to control both `kPreferredOutputBatchRows` and `kMaxOutputBatchRows`.
```c++
const std::string kSparkBatchSize = "spark.sql.execution.arrow.maxRecordsPerBatch";
```

Reply:
> With above logic, below Spark config is used to control both `kPreferredOutputBatchRows` and `kMaxOutputBatchRows`.
> 
> ```c++
> const std::string kSparkBatchSize = "spark.sql.execution.arrow.maxRecordsPerBatch";
> ```

this config actually control the pyspark's arrow batch size.

Reply:
Yes, batch size is configurable. we use `spark.sql.execution.arrow.maxRecordsPerBatch` in our jenkens job for tpcds/tpch.

Reply:
Created a pr to  change spark.gluten.sql.columnar.shuffleSplitDefaultSize to spark.gluten.sql.columnar.maxBatchSize in https://github.com/oap-project/gluten/pull/1542.